snapcraft/internal/lxd.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, *, output, source, project_options,
         _verify_remote(remote)
         self._container_name = '{}:snapcraft-{}'.format(remote, container_name)
         # Use the server architecture to avoid emulation overhead
-        kernel = self._get_remote_info()['environment']['kernel_architecture']
+        kernel = self._get_remote_info()['environment']['architectures'][0]
```

Comment:
I know because that's what LXD does in the source code. But I have no strong opinion one way or the other, so I'm happy to go for the fallback instead.